File: Lesson12_Inheritance/01_Reptile/MyExample.java
/*
Copyright Ann Barcomb and Emily Marasco, 2023
Licensed under GPL v3
See LICENSE.txt for more information.
*/
package edu.ucalgary.oop;

class Animal {
	private int age;

	public Animal(int age) {
		this.setAge(age);
	}

	public Animal() {
	}

	public int getAge() {
		return this.age;
	}

	public void setAge(int age) {
		this.age = age;
	}

	public void eat() {
		System.out.println("Nom nom nom");
	}
}

class Reptile extends Animal {
	private String scales;

	public Reptile(int age) {
		super(age);
	}

	public Reptile(int age, String scales) {
		super(age);
		setScales(scales);
	}

	
	public String getScales() {
		return this.scales;
	}

	public void setScales(String scales) {
		this.scales = scales;
	}
}

public class MyExample {
	public static void  main(String[] args) {
		Reptile chameleon = new Reptile(2, "multi-hued");
		Reptile gecko = new Reptile(1);

		System.out.println("Chameleons are " +
			chameleon.getScales() + " and " +
			"this one is " + chameleon.getAge() +
			" years old.");

		System.out.println("This gecko is " +
			gecko.getAge() + " years old and is" +
			" enjoying a meal.");
		gecko.eat();

		// We can also still use the Animal class without
		// using the Reptile class.
		Animal horse = new Animal(7);
		System.out.println("Our horse is " +
			horse.getAge() + " years old.");
	}
}

File: Lesson12_Inheritance/01_Reptile/MyExample.py
"""
Copyright Khawla Shnaikat and ChatGPT 3.5, 2023
Licensed under GPL v3
See LICENSE.txt for more information.
"""
class Animal:
    def __init__(self, age=0):
        self.age = age

    def get_age(self):
        return self.age

    def set_age(self, age):
        self.age = age

    def eat(self):
        print("Nom nom nom")

class Reptile(Animal):
    def __init__(self, age, scales=''):
        super().__init__(age)
        self.scales = scales

    def get_scales(self):
        return self.scales

    def set_scales(self, scales):
        self.scales = scales

def main():
    chameleon = Reptile(2, "multi-hued")
    gecko = Reptile(1)

    print(f"Chameleons are {chameleon.get_scales()} and this one is {chameleon.get_age()} years old.")
    print(f"This gecko is {gecko.get_age()} years old and is enjoying a meal.")
    gecko.eat()

    horse = Animal(7)
    print(f"Our horse is {horse.get_age()} years old.")

if __name__ == "__main__":
    main()
